uarray/backend.py

```diff
@@ -174,12 +174,25 @@ def replace_dispatchables(
         args, kwargs = argument_replacer(args, kwargs, tuple(replaced_args))
 
         kwargs = {k: kwargs[k] for k in opts if k in kwargs}
-        for k, v in defaults.items():
+        for k, v in kw_defaults.items():
             if k in kwargs and kwargs[k] is v:
                 del kwargs[k]
 
         return args, kwargs
 
+    def canonicalize_args(args):
+        if len(args) > len(arg_defaults):
```

Comment:
Isn't this branch already handled by the `return` at the end of this function?

Reply:
The issue is with this line
```python
for a, d in zip(args[::-1], arg_defaults[len(args) - 1 :: -1]):
```
the slice of `arg_defaults` will not match up with `args`.